lib/rules/function-name-format.js

```diff
@@ -0,0 +1,67 @@
+// Note that this file is nearly identical to mixin-name-format.js, placeholder-name-format.js, and variable-name-format.js
+'use strict';
+
+var helpers = require('../helpers');
+
+module.exports = {
+  'name': 'function-name-format',
+  'defaults': {
+    'allow-leading-underscore': true,
+    'convention': 'hyphenatedlowercase',
+    'convention-explanation': false
+  },
+  'detect': function (ast, parser) {
+    var result = [];
+
+    ast.traverseByType('function', function (node) {
+      var name = node.first('ident').content,
+          strippedName,
+          violationMessage = false;
+
+      strippedName = name;
+
+      if (parser.options['allow-leading-underscore'] && name[0] === '_') {
+        strippedName = name.slice(1);
+      }
+
+      switch (parser.options.convention) {
+      case 'hyphenatedlowercase':
```

Comment:
Although `npm test` and eslint aren't flagging this as an error technically these cases and default should be indented within the switch.